bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/NetworkHandler.java

```diff
@@ -153,69 +181,82 @@ public void finalDetectionResult(PresenceDetectionValue value) {
         }
 
-        this.retries = handlerConfiguration.retry.intValue();
-        presenceDetection.setRefreshInterval(Duration.ofMillis(handlerConfiguration.refreshInterval));
-        presenceDetection.setTimeout(Duration.ofMillis(handlerConfiguration.timeout));
-
-        wakeOnLanPacketSender = new WakeOnLanPacketSender(handlerConfiguration.macAddress,
-                handlerConfiguration.hostname, handlerConfiguration.port, handlerConfiguration.networkInterfaceNames);
+        this.retries = config.retry.intValue();
+        presenceDetection.setTimeout(Duration.ofMillis(config.timeout));
+        synchronized (this) {
+            this.presenceDetection = presenceDetection;
+            wakeOnLanPacketSender = new WakeOnLanPacketSender(config.macAddress, config.hostname, config.port,
+                    config.networkInterfaceNames);
+            if (config.refreshInterval > 0) {
+                refreshJob = executor.scheduleWithFixedDelay(presenceDetection::refresh, 0, config.refreshInterval,
+                        TimeUnit.MILLISECONDS);
+            }
+        }
 
         updateStatus(ThingStatus.ONLINE);
-        presenceDetection.startAutomaticRefresh();
-
-        updateNetworkProperties();
     }
 
     private void updateNetworkProperties() {
         // Update properties (after startAutomaticRefresh, to get the correct dhcp state)
         Map<String, String> properties = editProperties();
-        properties.put(NetworkBindingConstants.PROPERTY_ARP_STATE, presenceDetection.getArpPingState());
-        properties.put(NetworkBindingConstants.PROPERTY_ICMP_STATE, presenceDetection.getIPPingState());
-        properties.put(NetworkBindingConstants.PROPERTY_PRESENCE_DETECTION_TYPE, "");
-        properties.put(NetworkBindingConstants.PROPERTY_DHCP_STATE, presenceDetection.getDhcpState());
+        synchronized (this) {
+            PresenceDetection pd = presenceDetection;
+            if (pd == null) {
+                logger.warn("Can't update network properties because presenceDetection is null");
+                return;
+            }
+            properties.put(NetworkBindingConstants.PROPERTY_ARP_STATE, pd.getArpPingState());
+            properties.put(NetworkBindingConstants.PROPERTY_ICMP_STATE, pd.getIPPingState());
+            properties.put(NetworkBindingConstants.PROPERTY_PRESENCE_DETECTION_TYPE, "");
+            properties.put(NetworkBindingConstants.PROPERTY_DHCP_STATE, pd.getDhcpState());
+        }
         updateProperties(properties);
     }
 
     // Create a new network service and apply all configurations.
     @Override
     public void initialize() {
-        initialize(new PresenceDetection(this, scheduler,
-                Duration.ofMillis(configuration.cacheDeviceStateTimeInMS.intValue())));
+        executor.submit(() -> {
+            initialize(new PresenceDetection(this, Duration.ofMillis(configuration.cacheDeviceStateTimeInMS.intValue()),
+                    executor));
+        });
     }
```

Comment:
Not ideal as now the `initialize` method will now return without setting a state for the thing.

Reply:
You could set the thing status to UNKNOWN just before your executor.submit

Reply:
Yes, thanks!

Reply:
True, it should probably set `updateStatus(ThingStatus.UNKNOWN);` before calling the executor.